src/main/java/christmas/controller/ChristmasController.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    public void run() {
+        ExpectedVisitDate date = getExpectedVisitDate();
+        Orders orders = getOrders();
+        outputView.printDate(date.getDate().getDate());
+        BeforeEventPrice beforeEventPrice = getBeforeEventPrice(orders);
+        printGift(beforeEventPrice);
+        Benefits benefits = getBenefits(orders, date);
+        BenefitSum benefitSum = getBenefitSum(benefits);
+        printAfterEventPrice(beforeEventPrice, benefits);
+        printBadge(benefitSum);
+    }
+
+    private ExpectedVisitDate getExpectedVisitDate() {
+        return date();
+    }
+
+    private Orders readOrders(String orders) {
+        return new Orders(orders);
+    }
+
+    private ExpectedVisitDate readDate(int date) {
+        return new ExpectedVisitDate(date);
+    }
+
+    private BeforeEventPrice getBeforeEventPrice(Orders orders) {
+        BeforeEventPrice beforeEventPrice = new BeforeEventPrice(orders);
+        printBeforeEventPrice(beforeEventPrice);
+        return beforeEventPrice;
+    }
+
+    private void printGift(BeforeEventPrice beforeEventPrice) {
+        Gift gift = new Gift(beforeEventPrice.getBeforePrice(), Menu.CHAMPAGNE.getName(), GIFT_GIVEN_COUNT.getNumber());
+        printGift(gift);
+    }
+
+    private Benefits getBenefits(Orders orders, ExpectedVisitDate date) {
+        Benefits benefits = new Benefits(orders.toDtoList(), date);
+        printBenefits(benefits);
+        return benefits;
+    }
+
+    private BenefitSum getBenefitSum(Benefits benefits) {
+        BenefitSum benefitSum = new BenefitSum(benefits);
+        printBenefitSum(benefitSum);
+        return benefitSum;
+    }
+
+    private void printAfterEventPrice(BeforeEventPrice beforeEventPrice, Benefits benefits) {
+        AfterEventPrice afterEventPrice = new AfterEventPrice(beforeEventPrice, benefits);
+        printAfterEventPrice(afterEventPrice);
+    }
+
+    private void printBadge(BenefitSum benefitSum) {
+        Badge badge = getBadge(benefitSum.getBenefitSum().getSum());
+        printBadge(badge);
+    }
+
+    private void printGift(Gift gift) {
+        if (gift.getCount().getCount() <= 0) {
+            outputView.printGift(GIFT_NOTHING.getString());
+        }
+        if (gift.getCount().getCount() > 0) {
+            outputView.printGift(PRINT_ORDER_FORMAT.getString().formatted(
+                    MessageFormat.format(ORDER_PRINT_FORMAT.getString(), gift.getName().getName(),
+                            gift.getCount().getCount())));
+        }
+    }
+
+    private void printBeforeEventPrice(BeforeEventPrice beforeEventPrice) {
+        outputView.printBeforePrice(beforeEventPrice.getBeforePrice());
+    }
+
+    private void printBenefitSum(BenefitSum benefitSum) {
+        int sum = ZERO.getNumber();
+        if (benefitSum.getBenefitSum().getSum() > ZERO.getNumber()) {
+            sum = -benefitSum.getBenefitSum().getSum();
+        }
+        outputView.printBenefitSum(sum);
+    }
```

Comment:
음수로 바꾸는 책임을 benefitSum 내부로 옮기면 좋을 것 같아요. 새로운 메소드를 만들어도 좋구요.
```suggestion
    private void printBenefitSum(BenefitSum benefitSum) {
        outputView.printBenefitSum(benefitSum.getDiscounts());
    }
```